File: 프로그래머스/lv0/120890. 가까운 수/가까운 수.py
def solution(array, n):
    value = sorted(array, key=lambda x:abs(n-x))
    if abs(value[0] - n) == abs(value[1] - n):
        return min(value[0], value[1])
    return sorted(array, key=lambda x:abs(n-x))[0]

File: 프로그래머스/lv0/120890. 가까운 수/가까운 수.js
// function solution(array, n) {
//     let obj = {};
//     // 두 수의 차를 value로
//     array.forEach(item => {
//         obj[item] = Math.abs(item - n);
//     })
//     const minNum = Math.min(...Object.values(obj));
//     return parseInt(Object.keys(obj).find(key => obj[key] === minNum));
// }

// sol.2
function solution(array, n) {
    array.sort((a, b) => a - b);
    let diff = Infinity;
    let ind = Infinity;
    for (let i in array) {
        let calc = Math.abs(n - array[i]);
        if (calc < diff) {
            diff = calc;
            ind = i;
        }
    }
    return array[ind];
}